gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/MRCompactor.java

```diff
@@ -788,6 +792,13 @@ protected void afterExecute(Runnable r, Throwable t) {
             // Set the dataset status to COMPACTION_COMPLETE if compaction is successful.
             jobRunner.getDataset().setState(COMPACTION_COMPLETE);
           }
+          if (MRCompactor.this.compactorListener.isPresent()) {
```

Comment:
This seems to allow only one listener. Can you make it possible to have multiple listeners?